### ISSUE TITLE
google charts not showing up on any mobile browser

### ISSUE THREAD

**Issue by annongitter** [google/google-visualization-issues#2912]
my charts do not show up on mobile browsers but works totally fine (and is awesome) on all desktop browsers. 

My JS looks like so:

```
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
<script type="text/javascript">
  google.charts.load('current', {'packages':['bar']});
  google.charts.setOnLoadCallback(drawChart);

  function drawChart() {

    var data = google.visualization.arrayToDataTable([
    ### populate data here
    ]);

    var options = {
      chart: {
        title: 'Breakdown of VarA:',
        subtitle: 'VarA, VarA1, VarA2, VarA3 and VarA4',
      },
      bars: 'horizontal', // Required for Material Bar Charts.
      legend: { position: 'none', alignment: 'start' },
      vAxis: { gridlines: { count: 4 } },
      hAxis: {format: 'decimal', gridlines: { count: 4 } },
    };

    var chart = new google.charts.Bar(document.getElementById('barchart_material'));

    chart.draw(data, google.charts.Bar.convertOptions(options));
  }
</script>

```

my HTML looks like so:

```
<div class="col-sm-7">
<table class="table">
    <tbody>
      <tr>
        <td><div id="barchart_material"></div></td>
      </tr>
     </tbody>
  </table>
..snip...
```

I am perplexed why it wouldnt display on mobiles :(

**Comment by dlaliberte** [comment]
jquery 1.12.4 is very old.  I wouldn't be surprised if there are conflicts with that version and polyfills for various versions of browsers.  Any chance you could update that?

Also, please try to load Google Charts 'upcoming', which is now at version 51.  And try other versions as well, in particular '49'.

If polyfill conflicts are the problem, we may have a solution (in the future) to isolate polyfills in Google Charts.  I'm also concerned that web sites that now depend on the polyfills that Google Charts is providing will then suddenly stop working, so this may not be a trivial change that always works better.

**Comment by annongitter** [comment]
Thank you @dlaliberte for all the issues you pointed out. I have now upgraded my jquery to the latest but I still have the same issues (I built this app in 2017, so yea I admit its a bit old). I tried the different versions you had suggested but with no luck. However, I did see that in some rare instances it was rendering the plot on mobile (I have a tonne of pages in this app). Very bizzare. However, it ALWAYS works perfectly in firefox web browser.

**Comment by dlaliberte** [comment]
If it works sometimes but not always, then there are probably load-order dependencies.  And because of that, I am leaning more toward the problem being polyfill incompatibilities. 

If you have other libraries being loaded in the same page, the first thing to do would be to try to simplify the page as much as possible, eliminating everything you can until it hopefully starts working again.  Then add things back in until it fails again.  You can also try upgrading any of the other libraries you are using, but do that carefully.

**Comment by annongitter** [comment]
@dlaliberte You were totally right - it was a problem with Disqus js embeds. I removed this, used `49` version of google charts and boom everything works as expected. Unreal!

For sure back in 2017 when I had set it up, it all worked fine together - I guess something got upgraded along the way and this does not play nice with google charts anymore.

Thank you @dlaliberte for suggesting all these things to try it - I would have never imagined it was a load order problem. It was very bizzare when it would load at times and wouldnt load other times.
